IS1001Test.py

```diff
@@ -12,12 +12,356 @@
+                'type': 'string',
+                'description': 'Refresh Token to be used to obtain further Access Tokens'
+              },
+            },
+            'required': ['access_token', 'expires_in', 'token_type'],
+            'type': 'object',
+            'description': 'OAuth2 Response for the request of a Bearer Token',
+          },
+          400: {
+            'title': 'Token Error Response',
+            '$schema': 'http://json-schema.org/draft-07/schema#',
+            'properties': {
+              'error': {
+                'type': 'string',
+                'description': 'Error Type'
+              },
+              'error_uri': {
+                'type': 'string',
+                'description': 'A URI identifying a human-readable web page with information about the error'
+              },
+              'error_description': {
+                'type': 'string',
+                'description': 'Human-readable ASCII text providing additional information'
+              }
+            },
+            'required': ['error'],
+            'type': 'object',
+            'minItems': 1,
+            'description': 'Object defining error type and description',
+          }
+        },
+      }],
+    }
+  }
+
+  Example "apis" Object:
+  {
+      'auth': {
+          'raml': 'AuthorizationAPI.raml',
+          'version': 'v1.0',
+          'hostname': 'example.co.uk',
+          'port': 80,
+          'spec_branch': 'v1.0-dev',
+          'spec_path': 'cache/is-10',
+          'base_url': 'http://example.co.uk:80',
+          'name': 'Authorization API',
+          'ip': '127.0.1.1',
+          'url': 'http://example.co.uk:80/x-nmos/auth/v1.0/',
+          'spec': <Specification.Specification object at 0x7f5b3b0719b0>
+      }
+  }
+  """
+
     def __init__(self, apis):
         GenericTest.__init__(self, apis)
+
+        self.url = self.apis[AUTH_API_KEY]["url"]
+        self.bearer_token = None
+        self.client_data = None
+
+    def set_up_tests(self):
+        try:
+            signup_data = {
+                'username': AUTH_USERNAME,
+                'password': AUTH_PASSWORD,
+                'is04': 'read',
+                'is05': 'write'
+            }
+            status, response = self.do_request(
+                method="POST", url=self.url + 'signup', data=signup_data
+            )
+
+            if status is not True:
+                raise Exception
+
+        except Exception:
+            print("""
+                Ensure a User is already set-up on the Authorization Server that corresponds
+                to the 'AUTH_USERNAME' and 'AUTH_PASSWORD' config options
+            """)
+
+    def tear_down_tests(self):
+        try:
+            self._make_auth_request(method="GET", url_path='delete_client/' + self.client_data.client_id)
+        except Exception:
+            pass
+
+    def _make_auth_request(self, method, url_path, data=None, auth=(AUTH_USERNAME, AUTH_PASSWORD)):

```

Comment:
The auth tuple here may be a case of https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments. It probably won't matter in this case, but may be better to define it as None and then do an 'if auth is None' assign the defaults.

Reply:
As I'm not mutating the auth paramater inside the function it should be alright